realis_e2e_test.go

```diff
@@ -327,52 +326,60 @@ func TestRealisClient_CreateService_WithPulse_Thermos(t *testing.T) {
 		Limit: 1,
 	}
 
-	start := time.Now()
-	for i := 0; i*int(pulse) <= timeout; i++ {
+	var updateDetails []*aurora.JobUpdateDetails
 
-		fmt.Println("sending PulseJobUpdate....")
-		resp, err = r.PulseJobUpdate(result.GetKey())
-		assert.NotNil(t, resp)
-		assert.Nil(t, err)
+	ticker := time.NewTicker(time.Second * 3)
+	timer := time.NewTimer(time.Minute * 6)
+	defer ticker.Stop()
+	defer timer.Stop()
 
-		respDetail, err := r.JobUpdateDetails(updateQ)
-		assert.Nil(t, err)
+pulseLoop:
+	for {
+		select {
+		case <-ticker.C:
 
-		updateDetail := response.JobUpdateDetails(respDetail)
-		if len(updateDetail) == 0 {
-			fmt.Println("No update found")
-			assert.NotEqual(t, len(updateDetail), 0)
-		}
-		status := updateDetail[0].Update.Summary.State.Status
 
-		if _, ok := aurora.ACTIVE_JOB_UPDATE_STATES[status]; !ok {
+			fmt.Println("sending PulseJobUpdate....")
+			resp, err = r.PulseJobUpdate(result.GetKey())
+			assert.NotNil(t, resp)
+			assert.Nil(t, err)
```

Comment:
If you want to stop the execution if there is error, use `require` instead of `assert`, otherwise you will end up having a lot of assertion error caused by nil pointer in one of failed assert. see https://godoc.org/github.com/stretchr/testify/require

> Package require implements the same assertions as the `assert` package but stops test execution when a test fails.

Reply:
Didn't know this package existed. Thanks for pointing it out. Fixed!